iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/PhaseTapChangerAdderImpl.java

```diff
@@ -174,30 +176,37 @@ public StepAdder beginStep() {
     @Override
     public PhaseTapChanger add() {
         if (tapPosition == null) {
-            throw new ValidationException(transformer, "tap position is not set");
+            throw new ValidationException(parent, "tap position is not set");
         }
         if (steps.isEmpty()) {
-            throw new ValidationException(transformer, "a phase tap changer shall have at least one step");
+            throw new ValidationException(parent, "a phase tap changer shall have at least one step");
         }
         int highTapPosition = lowTapPosition + steps.size() - 1;
         if (tapPosition < lowTapPosition || tapPosition > highTapPosition) {
-            throw new ValidationException(transformer, "incorrect tap position "
+            throw new ValidationException(parent, "incorrect tap position "
                     + tapPosition + " [" + lowTapPosition + ", "
                     + highTapPosition + "]");
         }
-        ValidationUtil.checkPhaseTapChangerRegulation(transformer, regulationMode, regulationValue, regulating, regulationTerminal, getNetwork());
+        ValidationUtil.checkPhaseTapChangerRegulation(parent, regulationMode, regulationValue, regulating, regulationTerminal, getNetwork());
         if (!Double.isNaN(targetDeadband) && targetDeadband < 0) {
-            throw new ValidationException(transformer, "Unexpected value for target deadband of phase tap changer: " + targetDeadband);
+            throw new ValidationException(parent, "Unexpected value for target deadband of phase tap changer: " + targetDeadband);
         }
         PhaseTapChangerImpl tapChanger
-                = new PhaseTapChangerImpl(transformer, lowTapPosition, steps, regulationTerminal, tapPosition, regulating, regulationMode, regulationValue, targetDeadband);
-
-        Set<TapChanger> tapChangers = new HashSet<TapChanger>();
-        tapChangers.addAll(transformer.getAllTapChangers());
-        tapChangers.remove(transformer.getPhaseTapChanger());
-        ValidationUtil.checkOnlyOneTapChangerRegulatingEnabled(transformer, tapChangers, regulating);
+                = new PhaseTapChangerImpl(parent, lowTapPosition, steps, regulationTerminal, tapPosition, regulating, regulationMode, regulationValue, targetDeadband);
+
+        Set<TapChanger> tapChangers = new HashSet<>();
+        tapChangers.addAll(parent.getAllTapChangers());
+        tapChangers.remove(parent.getPhaseTapChanger());
+        ValidationUtil.checkOnlyOneTapChangerRegulatingEnabled(parent, tapChangers, regulating);
+
+        if (parent.getTransformer() instanceof ThreeWindingsTransformer) {
+            Set<TapChanger> tapChangersOnLeg = new HashSet<>();
+            tapChangersOnLeg.addAll(parent.getAllTapChangersOnLeg());
+            tapChangersOnLeg.remove(parent.getPhaseTapChanger());
+            ValidationUtil.checkOnlyOneTapChangerInThreeWindingsTranformer(parent, tapChangersOnLeg);
+        }
```

Comment:
Rather than doing that, with a boolean `hasRtc()` method on `TapChangerParent`, you can do:
```java
if (parent.getTransformer() instanceof ThreeWindingsTransformer && parent.hasRtc()) {
  LOGGER.warn("{} has more than one tap changer", parent);
}
```

Reply:
Done